x/stakeibc/keeper/reward_converter.go

```diff
@@ -57,75 +57,95 @@ type FeeInfo struct {
-	totalFeesAmount := sdk.NewDecFromInt(rewardAmount).Mul(strideFeePct).TruncateInt()
-	reinvestAmount := rewardAmount.Sub(totalFeesAmount)
-
 	// Get the rebate info from the host zone if applicable
-	// If there's no rebate, return 0 rebate and reinvest the full reward
+	// If there's no rebate, return 0 rebate and the full reward as the remainder
 	rebateInfo, chainHasRebate := hostZone.SafelyGetCommunityPoolRebate()
 	if !chainHasRebate {
-		feeInfo = FeeInfo{
-			RebateAmount:    sdkmath.ZeroInt(),
-			StrideFeeAmount: totalFeesAmount,
-			ReinvestAmount:  reinvestAmount,
-		}
-		return feeInfo, nil
+		return sdkmath.ZeroInt(), rewardAmount, nil
 	}
 
+	// Get the fee percentage from params
+	strideFee := sdk.NewIntFromUint64(k.GetParams(ctx).StrideCommission)
+	strideFeePercentage := sdk.NewDecFromInt(strideFee).Quo(sdk.NewDec(100))
+
 	// It shouldn't be possible to have 0 delegations (since there are rewards and there was a community pool stake)
 	// This will also prevent a division by 0 error
 	if hostZone.TotalDelegations.IsZero() {
-		return feeInfo, errorsmod.Wrapf(types.ErrDivisionByZero,
+		return sdkmath.ZeroInt(), sdkmath.ZeroInt(), errorsmod.Wrapf(types.ErrDivisionByZero,
 			"unable to calculate rebate amount for %s since total delegations are 0", chainId)
 	}
 
 	// It also shouldn't be possible for the liquid stake amount to be greater than the full TVL
 	if rebateInfo.LiquidStakeAmount.GT(hostZone.TotalDelegations) {
-		return feeInfo, errorsmod.Wrapf(types.ErrFeeSplitInvariantFailed,
+		return sdkmath.ZeroInt(), sdkmath.ZeroInt(), errorsmod.Wrapf(types.ErrFeeSplitInvariantFailed,
 			"community pool liquid staked amount greater than total delegations")
 	}
 
 	// The rebate amount is determined by the contribution of the community pool stake towards the total TVL,
 	// multiplied by the rebate fee percentage
 	contributePercentage := sdk.NewDecFromInt(rebateInfo.LiquidStakeAmount).Quo(sdk.NewDecFromInt(hostZone.TotalDelegations))
-	rebateAmount := sdk.NewDecFromInt(totalFeesAmount).Mul(contributePercentage).Mul(rebateInfo.RebatePercentage).TruncateInt()
-	strideFeeAmount := totalFeesAmount.Sub(rebateAmount)
+	totalFeesAmount := sdk.NewDecFromInt(rewardAmount).Mul(strideFeePercentage).TruncateInt()
+	rebateAmount = sdk.NewDecFromInt(totalFeesAmount).Mul(contributePercentage).Mul(rebateInfo.RebatePercentage).TruncateInt()
+	remainingAmount = rewardAmount.Sub(rebateAmount)
 
-	// Sanity check calculation
-	if !rebateAmount.Add(strideFeeAmount).Add(reinvestAmount).Equal(rewardAmount) {
-		return feeInfo, errorsmod.Wrapf(types.ErrFeeSplitInvariantFailed, "fee info compontents do not sum to rewards")
-	}
+	return rebateAmount, remainingAmount, nil
+}
 
-	feeInfo = FeeInfo{
-		RebateAmount:    rebateAmount,
-		StrideFeeAmount: strideFeeAmount,
-		ReinvestAmount:  reinvestAmount,
+// Given the reward balance from an ICQ, calculates the portions intended as a stride fee vs for reinvestment
```

Comment:
suggestion for clarity:


Given the balance of **host-denom** rewards from an ICQ, calculate the portions intended as a stride fee vs for reinvestment

*note: `CalculateRewardsFeeSplit` is run on **host-denom** rewards. In the case of DYDX this is run **_after_** rewards are pushed through the trade route.*

Reply:
done

Reply:
lgtm